openapi/workflow_execution_service.swagger.yaml

```diff
@@ -75,24 +75,24 @@ paths:
         - name: page_size
           description: |-
             OPTIONAL
-            Number of workflows to return in a page.
+            The preferred number of workflow submissions to return in a page.
+            The actual number of items returned is implementation dependent.
           in: query
           required: false
           type: integer
           format: int64
         - name: page_token
           description: |-
             OPTIONAL
-            Token to use to indicate where to start getting results. If unspecified, returns the first
+            Token to use to indicate where to start getting results. If unspecified, return the first
             page of results.
           in: query
           required: false
           type: string
         - name: tag_search
           description: |-
             OPTIONAL
-            For each key, if the key's value is empty string then match workflows that are tagged with
-            this key regardless of value.
+            Only return workflow submissions where this key is present in "tags" (regardless of tag value).
```

Comment:
making sure I understand:
a) we let you search for "key K exists", but don't provide any way to search for "key K has value V"
b) we let you search for a single key, but don't provide any way to search for "key J and/or K exist"
Is that right? If so, it seems like a reasonable place to start, but I could be talked into doing more or less depending on what our initial driver projects need.

Reply:
I don't have strong feelings about this, I was just trying to reword it to make it clearer what behavior was being specified.

Reply:
I don't care strongly either way but without claiming to speak for @mckinsel - what we've seen from the group he works with using our stuff is they want both K and K/V pair as K is sometimes used as a straight tag and K/V pair is used as a, well, K/V pair.

Reply:
If we need to support both search for "K exists" and "K with value" exists, then I think tag_search needs to allow specification of that. For example, tag_search is structure that contains "key" field (required if tag_search is specified) and "value" field (optional; if not present we only check key existence.

From my naive point of view, this seems a bit ornate. I'd like to understand the use case for "key + value" and why a simple set of string tags is not sufficient.

Reply:
It'd also be helpful to clarify how these tags are related (if at all) to the "tags" field of [`WorkflowRequest`](https://github.com/ga4gh/workflow-execution-service-schemas/blob/27b1701206d7c7171d0dc77eb7c4d4a35c3e6953/openapi/workflow_execution_service.swagger.yaml#L435-L487):

```
    tags:
        type: object
        additionalProperties:
          type: string
        title: |-
          OPTIONAL
          A key-value map of arbitrary metadata outside the scope of the workflow_params but useful to track with this workflow request
```

Reply:
Meta comment: user kbergin made a comment and it’s been listed as pending for a few hours. Who can unblock this?

Reply:
@tetron do you have the ability to approve the comment from @kbergin ?

Reply:
https://help.github.com/articles/reviewing-proposed-changes-in-a-pull-request/

> Before you submit your review, your line comments are pending and only visible to you.

That might be what's going on?

Reply:
Oh I bet you’re right. I was going to say, I’d never seen a setting requiring comment approvals.

Reply:
So sorry for the meta-comment confusion!